### PR TITLE
Rap 3052

### PR DESCRIPTION
**Pull request by scarabus**
First pass at upgrading Mongo to 3.0.2
This still uses the deprecated DB object.
Second pass will replace DB with MongoDatabase

### CODE REVIEW

File: Libs/RaptureAddins/MongoDb/src/main/java/rapture/blob/mongodb/MongoDBBlobStore.java [hunk 1]
```diff
@@ -24,22 +24,25 @@
...
 import rapture.common.exception.RaptureExceptionFactory;
 
-import com.mongodb.MongoException;
-
 public class MongoDBBlobStore extends BaseBlobStore {
     private final static String INSTANCE_NAME = "default";
     public static final String PRIMARY_CONFIG = "prefix";
     public static final String SECONDARY_CONFIG = "grid";
     public static final String MULTIPART = "multipart";
+	private Messages mongoMsgCatalog = new Messages("Mongo");
```
Comment (dukenguyen) [inline Libs/RaptureAddins/MongoDb/src/main/java/rapture/blob/mongodb/MongoDBBlobStore.java]: formatting is off, should be 4 spaces not 8.  applies to most changes in this PR.

**Comment by dukenguyen** [comment]
please fix all formatting issues so the real diff can be seen.  otherwise it shows every line as different.

File: Libs/RaptureAddins/MongoDb/src/main/java/rapture/mongodb/MongoDBFactory.java [hunk 2]
```diff
@@ -91,38 +100,32 @@ private Mongo getMongoFromLocalConfig(String instanceName) {
         log.info("Collection is " + uri.getCollection());
 
         try {
-            Mongo mongo = new MongoClient(uri);
-            DB db = mongo.getDB(uri.getDatabase());
-            db.authenticate(uri.getUsername(), uri.getPassword());
-            mongoDBs.put(instanceName, db);
+            MongoCredential credential = MongoCredential.createCredential(uri.getUsername(), uri.getDatabase(), uri.getPassword());
```
Comment (scarabus) [inline Libs/RaptureAddins/MongoDb/src/main/java/rapture/mongodb/MongoDBFactory.java]: Not sure if this is actually necessary. It seems to work without it.

File: Libs/RaptureAddins/MongoDb/src/main/java/rapture/mongodb/MongoDBFactory.java [hunk 3]
```diff
@@ -149,8 +155,31 @@ private DB _getDB(String instanceName) {
             if (mongoDBs.containsKey(instanceName)) {
                 return mongoDBs.get(instanceName);
             } else {
-                RaptureException raptException = RaptureExceptionFactory.create(HttpURLConnection.HTTP_INTERNAL_ERROR, "Not initialized");
-                throw raptException;
+                throw RaptureExceptionFactory.create(HttpURLConnection.HTTP_BAD_REQUEST, mongoMsgCatalog.getMessage("NotInitialized"));
+            }
+        }
+    }
+
```
Comment (scarabus) [inline Libs/RaptureAddins/MongoDb/src/main/java/rapture/mongodb/MongoDBFactory.java]: The following is next-gen stuff - DB is deprecated in favor of MongoDatabase.

**Pull request merged**